app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -76,6 +79,31 @@ def self.find_with_expiring_pools(_key, _operator, days_from_now)
         end
       end
 
+      def reset_katello_status
+        self.host_statuses.where("#{HostStatus::Status.table_name}.type like :query", query: "Katello%").each do |status|
+          if status.class.const_defined?(:UNKNOWN)
```

Comment:
Maybe we should have a requirement which states that all of our statuses have an UNKNOWN?

Reply:
Hmm how can we enforce that?

Reply:
in plugin.rb we hook into the statuses: https://github.com/Katello/katello/blob/1621c4f1c54253c58d5a27a88df512bbbb42d88f/lib/katello/plugin.rb#L269-L276

We could:

- somewhere (engine.rb?) define a 'statuses' public method to get a list of all the classes
- change plugin.rb to iterate over that to register the statuses
- call that same public method here in order to get the class names for this query (vs using like)
- adjust the test to create one of each of the status classes (this part will enforce that they all provide an UNKNOWN value)

thoughts?